### DISCUSSION TITLE
Feature request: quarto title page for docx

### DISCUSSION THREAD

**Discussion by mnlang** [quarto-dev/quarto-cli#11018] [Q&A]
Are there any plans to support title pages for docx in the near future?

As far as I understand, [this is not that straightforward](https://forum.posit.co/t/quarto-book-title-page-in-docx/161686), but since Microsoft Word is still widely used in many (large) companies where customized title pages are a must, this would greatly increase the usability of Quarto outside of academia.

**Comment by mcanouil** [comment]
FYI, you could try to make an extension to inject what Pandoc/Quarto does not.
See for example https://github.com/kapsner/authors-block

**Comment by Steinthal** [comment]
As a workaround based on "officedown" and "officer", I use an include that defines the title page and places the TOC after the title page .

Disadvantages -- especially for a larger numbers of files to be changed at once : 
- "officer" inserts the TOC as field. When opening the word file, you need to confirm to update fields.
- The content of  metadata like "title", "subtitle", and "author" is inserted at the beginning of the title page. Those lines have to be deleted manually.

The qmd-include refers to custom styles in the reference.docx. You can format the text on the title page using this custom styles as you like.
Quarto inserts metadata and parameters as document properties which can be used in header and footer lines of the reference.docx, e.g. on page 2ff.

Here's the basic structure of the qmd-include:

````
 ```{r}
#| label: load-officer
#| warning: false
#| message: false
#| echo: false
library(officedown)
library(officer)
```

:::: {.content-visible when-format="docx"}
::: {custom-style="my_title"}
{{< meta title >}}
:::

::: {custom-style="my_subtitle"}
{{< meta subtitle >}}
:::

{{< pagebreak >}}

::: {custom-style="my_TOC_Heading"}
{{< meta toc-title >}}
:::

```{r}
#| label: place-toc-in-Word
#| echo: false
block_toc()
```

{{< pagebreak >}}

::::
````

**Reply by mcanouil** [reply]
Indeed, those R packages produce Open XML code so they work inside Quarto.
Thanks for sharing!

**Comment by Liripo** [comment]
I tried using the code above, but the `toc` not include docx. 

my quarto version is `1.5.57`
``````qmd
---
format: 
  docx: default
---

```{r}
#| label: load-officer
#| warning: false
#| message: false
#| echo: false
library(officedown)
library(officer)
```

:::: {.content-visible when-format="docx"}
::: {custom-style="my_title"}
{{< meta title >}}
:::
  
::: {custom-style="my_subtitle"}
{{< meta subtitle >}}
:::
  
{{< pagebreak >}}

::: {custom-style="my_TOC_Heading"}
{{< meta toc-title >}}
:::
  
```{r}
#| label: place-toc-in-Word
#| echo: false
block_toc()
```

{{< pagebreak >}}

::::

## h1

### h2

text.
``````

**Reply by Steinthal** [reply]
Did you pay attention to this point:
- "officer" inserts the TOC as field. When opening the word file, you need to confirm to update fields.

I've just checked with your example. When I open the Word file, I'm notified that the file contains fields that need to be updated. If you cancel this step or don't confirm the next step to update the page numbers or the complete TOC, it is not included into the word file.

**Comment by Liripo** [comment]
@Steinthal Thanks for your help, I understand.
